mqtt/mqtt-bridge/src/pump/ingress.rs

```diff
@@ -33,10 +33,12 @@ where
     }
 
     /// Runs ingress processing.
-    pub(crate) async fn run(mut self) {
+    pub(crate) async fn run(mut self) -> Result<(), IngressError> {
```

Comment:
I'm guessing you foresee some reason to return Err in egress/ingress?

Reply:
yes, that's what we discussed with Anca. what should Pump do when the client errors out? proably a bit premature changes however I need them to return some error.

Reply:
is this going to shutdown the bridge? right now we don't recreate it so this will trigger the broker shutdown I think. I was not planning to recreate the bridge because we discussed about validation so that client doesn't return errors.